File: 2138. Divide a String Into Groups of Size k/solution.java
class Solution {
  public String[] divideString(String s, int k, char fill) {
    List<String> result = new ArrayList<String>();
    String word = "";
    for (int index = 0; index < s.length(); index++){
      Character letter = s.charAt(index);
      word += letter;
      if (word.length() == k){
        result.add(word);
        word = "";
      }
    }

    if (word.length() != 0){
      for (int index = word.length(); index < k; index++){
        word += fill;
        if (word.length() == k){
          result.add(word);
        }
      }
    }

    String[] finalResult = new String[result.size()];
    result.toArray(finalResult);
    return finalResult;
  }
}

File: 2138. Divide a String Into Groups of Size k/solution.cpp
class Solution {
public:
  vector<string> divideString(string s, int k, char fill) {
    vector<string> result;
    string word = "";
    for (int index = 0; index < s.size(); index++){
      char letter = s[index];
      word += letter;
      if (word.size() == k){
        result.push_back(word);
        word = "";
      }
    }

    if (word.size() != 0){
      for (int index = word.size(); index < k; index++){
        word += fill;
        if (word.size() == k){
          result.push_back(word);
        }
      }
    }

    return result;
  }
};